cushion/persist/mem.py

```diff
@@ -0,0 +1,120 @@
+
+import operator
+from uuid import uuid4
+
+import execjs
+
+from .base import BaseConnection
+
+
+mapwrap = '''
+function map_wrapper(doc, meta, outq) {
+    function emit(key, val) {
+        emit.outq.push([key, val, meta._id])
+    }
+    emit.outq = outq
+    var mapf = %MAPF%
+    mapf(doc, meta)
+    return emit.outq
+}
+'''
+
+
+class MemDoc(object):
+    def __init__(self, docid, value):
+        self.key = docid
+        self.value = value
+
+
+class MemResult(object):
+    def __init__(self, key, docid, value, doc=None):
+        self.key = key
+        self.docid = docid
+        self.value = value
+        self.doc = doc
+
+
+class MemResultSet(object):
+    def __init__(self, results, include_docs=False):
+        self._results = results
+        self.include_docs = include_docs
+
+    def __iter__(self):
+        for r in self._results:
+            yield r
+
+
+class MemConnection(BaseConnection):
+
+    def __init__(self):
+        self.designs = {}
+        self.data = {}
+
+    def get(self, key):
+        return self.data.get(key, None)
+
+    def set(self, key, value):
+        if key is None:
+            key = uuid4().hex
+        self.data[key] = value
+        return key, uuid4().hex # fake cas
+
+    def delete(self, key):
+        del self.data[key]
+
+    def query(self, design, name, **kw):
+        # **note** NO REDUCE YET
+        view_key = '/'.join((design, name))
+        if view_key not in self.designs:
+            raise Exception('view not found')
+        view = self.designs[view_key]
+        mapf_ctx = view.get('mapf')
+        redf_ctx = view.get('redf')
+        include_docs = False
+        if 'include_docs' in kw:
+            if kw['include_docs'] and kw['include_docs'] is not 'false':
```

Comment:
seems this could just be `if kw['include_docs'] is not 'false':`, right? you've already checked that its there, unless I'm missing something...